File: palindrome/palidrome.js
function isPalindrome(word) {
  if (typeof word !== 'string') {
    throw TypeError('Argument must be a string');
  }
  word = word.toLowerCase();
  return word === [...word].toReversed().join('');
}

console.log(isPalindrome('kajak')); // true
console.log(isPalindrome('test')); // false
console.log(isPalindrome(123)); // TypeError: Argument must be a string

File: palindrome/palidrome.py
def is_palidrome(word):
  if type(word) is not str:
    raise TypeError('Argument must be a string')
  word = word.lower()
  return word == word[::-1]

print(is_palidrome('kajak')) # True
print(is_palidrome('test')) # False
print(is_palidrome(123)) # TypeError: Argument must be a string